lib/frameworks/mocha.js

```diff
@@ -46,7 +47,7 @@ module.exports.run = function(config, specs, capabilities) {
     /**
      * enable generators if supported
      */
-    if(hasES6Support) {
+    if(hasES6Support && interfaces[config.mochaOpts.ui]) {
```

Comment:
This prevents the above changes from breaking anything for interfaces you don't support.